### ISSUE TITLE
Add track length data for all tracks, and modify publish process to gather that data -

### ISSUE THREAD

**Issue by mbrakken** [therabble/ccr-app#8]
Needs are across api, admin, and public app

**Comment by mbrakken** [comment]
removing this from Beta-done milestone

**Comment by mbrakken** [comment]
#28 is a piece of this

**Comment by lahollen** [comment]
@mbrakken can you give an update on this ticket?

**Comment by mbrakken** [comment]
I put together a jsfs pr that should allow me to access the response headers. I have a script mostly written to update all tracks, just waiting on the jsfs update.

A cursory look tells me that this won't resolve all cases. Files that were already in wav format don't appear to have duration headers, but those should become obvious.

**Comment by jjg** [comment]
@mbrakken  can you give me an example of a WAVE file that doesn't have the duration header?  I expect that they all should, but maybe I overlooked something.

**Comment by mbrakken** [comment]
Just for thread reference, I send @jjg relevant urls.

For posterity, short script:
```javascript
// Done in browser console
// $E is set to the global store via ember inspector, the Container tab
$E.all('review-and-publish-album').content.forEach(function(album){
    album.get('tracks').forEach(function(track){
        var url = track.get('url');
        var xhr = new XMLHttpRequest();
        xhr.open('HEAD', url);
        xhr.onreadystatechange = function(){
            if (xhr.readyState === 2) {
                track.set('duration', xhr.getResponseHeader('X-Media-Duration'))
            }
        }
        xhr.send(null);
    });
})
```

**Comment by mbrakken** [comment]
All tracks should have duration info now. We'll still want to update the upload process to capture that info at that time. The back end is now set up for that, but it should be a separate ticket. Closing.
